src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -227,7 +227,17 @@ private static Uri GetUriFromString(string value)
             else
             {
                 host = value.Substring(0, separatorIndex);
-                if (!ushort.TryParse(value.AsSpan(separatorIndex + 1), out port))
+                int endIndex = separatorIndex + 1;
+                // Strip any trailing characters after port num,ber.
+                while (endIndex < value.Length)
+                {
+                    if (!char.IsDigit(value[endIndex]))
```

Comment:
Note that char.IsDigit returns true for things other than '0'-'9'. It may not matter here other than for very esoteric cases.